### PR TITLE
Chapter 7

### PR DESCRIPTION
**Pull request by balqis277**
ready

### CODE REVIEW

**Comment by balqis277** [comment]
ready

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/users.scss:1]: please delete this file

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (anlvq-1058) [inline app/helpers/users_helper.rb:1]: please delete this file

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -2,19 +2,19 @@
...
+      <div class="alert alert-<%= message_type %>"><%= message %></div>
+    <% end %>
    <%= yield %>
-   </div>
-
-  </body>
-</html>
+  </div>
+ </body>
+</html>
```
Comment (anlvq-1058) [inline app/views/layouts/application.html.erb:20]: add  the last line blank and empty

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (anlvq-1058) [inline test/fixtures/users.yml:1]: delete this file, please

File: test/models/user_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (anlvq-1058) [inline test/models/user_test.rb:1]: delete this file, please

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: test/controllers/users_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (anlvq-1058) [inline test/controllers/users_controller_test.rb:1]: please delete this file

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: config/routes.rb [hunk 7]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   root 'static_pages#home'
   get 'static_pages/help'
-
+  get 'static_pages/contact'
   get "/signup", to: "users#new"
+  post "/signup", to: "users#create"
   resources :users
```
Comment (anlvq-1058) [inline config/routes.rb:7]:
```suggestion
  resources :users, except: %I(new create)
```

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -1,2 +1,30 @@
...
+            <%= t "the_form_contains" %>
+            <%= pluralize @user.errors.count, "error" %>.
+        </div>
+        <ul>
+          <% @user.errors.full_messages.each do |msg| %>
+            <li><%= msg %></li>
+          <% end %>
+        </ul>
+    </div>
+ <% end %>
```
Comment (anlvq-1058) [inline app/views/users/new.html.erb:18]: worng indent

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -1,2 +1,33 @@
-<h1>Users#new</h1>
-<p>Find me in app/views/users/new.html.erb</p>
+<% provide :title, t("sign_up") %>
+<h1>Sign up</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @user do |f| %>
+
+    <% if @user.errors.any? %>
```
Comment (huongnt-2545) [inline app/views/users/new.html.erb:7]: indent from line 7 -> 20

File: app/views/users/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<% provide :title, @user.name %>
+<div class="row">
+  <aside class="col-md-4">
+    <section class="user_info">
+     <h1>
```
Comment (huongnt-2545) [inline app/views/users/show.html.erb:5]: indent from line 5 -> 10

File: config/routes.rb [hunk 11]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   root 'static_pages#home'
   get 'static_pages/help'
-
+  get 'static_pages/contact'
   get "/signup", to: "users#new"
-  resources :users
+  post "/signup", to: "users#create"
+  resources :users, except: %I(new create)
```
Comment (huongnt-2545) [inline config/routes.rb:7]:
```suggestion
  resources :users, only: %i(new create)
```

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Review by huongnt-2545 — COMMENTED** [review]
Still comment is not fix yet, please check

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -2,19 +2,19 @@
...
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
-
-  <body>
-    <%= render "layouts/header" %>
+<body>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb:11]: wrong indent

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**